enum_test.go

```diff
@@ -134,19 +129,19 @@ func Test_Enum_Undefined(t *testing.T) {
 		RoleAdmin = enum.New[Role]("admin")
 	)
 
-	moderator := enum.EnumOf[Role]("moderator")
-	assert.NotEqual(t, moderator, RoleUser)
+	moderator, _ := enum.EnumOf[Role]("moderator")
+	//assert.NotEqual(t, moderator, RoleUser)
 	assert.NotEqual(t, moderator, RoleAdmin)
 
 	assert.True(t, enum.IsValid(RoleUser))
 	assert.True(t, enum.IsValid(RoleAdmin))
```

Comment:
I think you can adjust the test like this:

```suggestion
	assert.True(t, enum.IsValid(RoleUser))
	assert.True(t, enum.IsValid(RoleAdmin))

	_, ok := enum.EnumOf[Role]("moderator")
	assert.False(t, ok)
```